guardian/recipes-backend
# Recipes responder skeleton

## What does this change?

https://trello.com/c/xuSwPkfG

- importing the content-api-models npm package
- set up a new lambda function
- use this code (in the lambda function package) to start off unmarshalling and parsing the content: https://github.com/guardian/pubflow-two/blob/main/packages/lambda/crier-responder/src/main.ts
- (also in the lambda function package) supply stub definitions for handleContentUpdate, handleContentUpdateRetrievable, handleDeletedContent

## How to test

There's a basic test of the decoder in-line.  It's kinda hard to directly test because the functions that get called are all stubs which throw exceptions; however, you could deploy it to CODE and watch the lambda run.  Check the logs to see that the expected exceptions get called, then either disable the kinesis trigger on the lambda or delete it by rolling back

## How can we measure success?

In a position to start building out the backend functionality

## Have we considered potential risks?

n/a
